core/src/main/java/com/rudderstack/android/sdk/core/RudderDeviceModeManager.java

```diff
@@ -221,43 +226,58 @@ private void isDeviceModeFactoriesNotPresent() {
-            if (destinationsWithTransformations.isEmpty()) {
-                RudderLogger.logVerbose(String.format(Locale.US, "RudderDeviceModeManager: makeFactoryDump: Marking event with rowId %s as DEVICE_MODE_PROCESSING DONE as it has no device mode destinations with transformations", rowId));
+        synchronized (replayMessageQueueLock) {
+            if (this.areDeviceModeFactoriesAbsent) {
+                RudderLogger.logVerbose(String.format(Locale.US, "RudderDeviceModeManager: makeFactoryDump: Marking event with rowId %s as DEVICE_MODE_PROCESSING_DONE as device mode factories are absent", rowId));
                 dbPersistentManager.markDeviceModeDone(Arrays.asList(rowId));
-            } else {
-                for (String destinationName : destinationsWithTransformations) {
-                    RudderLogger.logDebug(String.format(Locale.US, "RudderDeviceModeManager: makeFactoryDump: Destination %s needs transformation, hence the event will be batched and sent to transformation service", destinationName));
+            } else if (areFactoriesInitialized || fromHistory) {
+                List<String> eligibleDestinations = getEligibleDestinations(message);
+
+                List<String> destinationsWithTransformations = getDestinationsWithTransformationStatus(TRANSFORMATION_STATUS.ENABLED, eligibleDestinations);
+                if (destinationsWithTransformations.isEmpty()) {
+                    RudderLogger.logVerbose(String.format(Locale.US, "RudderDeviceModeManager: makeFactoryDump: Marking event with rowId %s as DEVICE_MODE_PROCESSING_DONE as it has no device mode destinations with transformations", rowId));
+                    dbPersistentManager.markDeviceModeDone(Arrays.asList(rowId));
+                } else {
+                    for (String destinationName : destinationsWithTransformations) {
+                        RudderLogger.logDebug(String.format(Locale.US, "RudderDeviceModeManager: makeFactoryDump: Destination %s needs transformation, hence the event will be batched and sent to transformation service", destinationName));
+                    }
+                    preferenceManager.saveOffsetValue(++this.offset);
                 }
-            }
 
-            List<String> destinationsWithoutTransformations = getDestinationsWithTransformationStatus(TRANSFORMATION_STATUS.DISABLED, eligibleDestinations);
-            dumpEventToDestinations(message, destinationsWithoutTransformations, "makeFactoryDump");
+                List<String> destinationsWithoutTransformations = getDestinationsWithTransformationStatus(TRANSFORMATION_STATUS.DISABLED, eligibleDestinations);
+                dumpEventToDestinations(message, destinationsWithoutTransformations, "makeFactoryDump");
+            }
         }
     }
 
+    void updateOffsetValueAfterTransformation(int messageIdsSize) {
```

Comment:
where is this method used?

Reply:
This method will be required in the DMT to update the offset value. Currently, it is not used anywhere. 
LMK if you think pushing this change in the DMT PR is better. If that is the case, I'll remove it.